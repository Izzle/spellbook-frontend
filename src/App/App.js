import React, { Component } from 'react';
import { Route, Link } from 'react-router-dom';
import './App.css';

class App extends Component{

  state = {};

  componentDidMount() {

  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <div>Testing 123</div>
        </header>
      </div>
    );
  }

}

export default App;
