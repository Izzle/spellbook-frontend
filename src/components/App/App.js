import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import Header from '../Header/Header';
import LandingPage from '../../routes/LandingPage/LandingPage';
import BuildSpellBookPage from '../../routes/BuildSpellBookPage/BuildSpellBookPage';
import CreateSpellPage from '../../routes/CreateSpellPage/CreateSpellPage';
import LoginPage from '../../routes/LoginPage/LoginPage';
import NotFoundPage from '../../routes/NotFoundPage/NotFoundPage';
import RegisterPage from '../../routes/RegisterPage/RegisterPage';
import ResetPasswordPage from '../../routes/ResetPasswordPage/ResetPasswordPage';
import SpellBookListPage from '../../routes/SpellBookListPage/SpellBookListPage';
import SpellBookPage from '../../routes/SpellBookPage/SpellBookPage';
import SpellLibraryPage from '../../routes/SpellLibraryPage/SpellLibraryPage';
import SpellPage from '../../routes/SpellPage/SpellPage';
import SpellContext from '../../contexts/SpellContext';

import './App.css';


class App extends Component{

  state = { 
    hasError: false,
    spellbooks: [],
    spells: []
  };

  static getDerviedStateFromError(error) {
    console.log(error);
    return { hasError: true };
  }

  setSpellBookList = ( spellbooks ) => {
    this.setState({
      spellbooks
    });
  }

  setSpells = ( spells ) => {
    //console.log(spells);
    this.setState({
      spells
    });
  }

  addSpell = spell => {
    // Adds a spell to state / context
    this.setState([
      ...this.state.spells,
      spell
    ]);
  }

  componentDidMount() {

  }

  render() {
    const contextValue = {
      spellbooks: this.state.spellbooks,
      spells: this.state.spells,
      setSpellBookList: this.setSpellBookList,
      setSpells: this.setSpells
    };

    return (
      <SpellContext.Provider value={contextValue}>
        <div className='App'>
          <header className='App__header'>
            <Header />
          </header>
          <main className='App__main'>
            {this.state.hasError && <p className='red'>There was an error!</p>}
            <Switch>
              <Route 
                exact
                path={'/'}
                component={LandingPage}
              />
              <Route 
                path={'/login'}
                component={LoginPage}
              />
              <Route 
                path={'/register'}
                component={RegisterPage}
              />
              <Route 
                path={'/reset-password'}
                component={ResetPasswordPage}
              />
              <Route 
                path={'/spellbook'}
                component={SpellBookListPage}/>
              <Route 
                path={'/spellbook/:id'}
                component={SpellBookPage}/>
              <Route 
                path={'/spellbook-builder'}
                component={BuildSpellBookPage}
              />
              <Route 
                path={'/spell-creator'}
                component={CreateSpellPage}
              />
              <Route 
                path={'/spells'}
                component={SpellLibraryPage}
              />
              <Route 
                path={'/spells/:id'}
                component={SpellPage}
              />
              <Route 
                path={'/page-not-found'}
                component={NotFoundPage}
              />
            </Switch>
          </main>
        </div>
      </SpellContext.Provider>
    );
  }

}

export default App;
