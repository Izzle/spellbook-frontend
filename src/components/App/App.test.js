import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import App from './App';
import { MemoryRouter } from 'react-router-dom';


describe('<App />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<MemoryRouter><App /></MemoryRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders this UI as expected', () => {
        const tree = renderer.create(<MemoryRouter><App /></MemoryRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    });    
});