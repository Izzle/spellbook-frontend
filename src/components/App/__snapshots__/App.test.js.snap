// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders this UI as expected 1`] = `
<div
  className="App"
>
  <header
    className="App__header"
  >
    <nav
      className="Header"
    >
      <ul
        className="Nav__list"
      >
        <li>
          <a
            className="Nav__Link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            className="Nav__Link"
            href="/spellbook"
            onClick={[Function]}
          >
            View Spellbooks
          </a>
        </li>
        <li>
          <a
            className="Nav__Link"
            href="/spell-creator"
            onClick={[Function]}
          >
            Create a Spell
          </a>
        </li>
        <li>
          <a
            className="Nav__Link"
            href="/spells"
            onClick={[Function]}
          >
            Spells
          </a>
        </li>
      </ul>
      <h1>
        <a
          href="/"
          onClick={[Function]}
        >
          Spellbook
        </a>
      </h1>
    </nav>
  </header>
  <main
    className="App__main"
  >
    <section
      className="LandingPage"
    >
      <h3>
        What is SpellBook?
      </h3>
      <p>
        It's a tool to keep track of your spells in Dungeons and Dragons. You can view and read up on spells, create your own custom spells, and add spells to different spellbooks for each character you play.
      </p>
      <h3>
        How do I make a spellbook?
      </h3>
      <p>
        First select 'View Spellbooks' then select the book you would like to view. From there, you can click 'Edit Spellbook' to see all available spells. Any spells currently in your Spellbook will be highlighted, just click on any spell to add or remove it.
      </p>
      <p>
        When you are done making your selections, click 'Save Spellbook' to save your spellbook for future use.
      </p>
      <h3>
        How do I make a new spell?
      </h3>
      <p>
        Click on 'Create a Spell' in the navigation, fill out all information about the spell such as its name, range, and duration. When you're done, hit 'Create Spell' and your spell will be added to all the spells in our library. From then on, you can add your custom spell to any spellbook!
      </p>
    </section>
  </main>
</div>
`;
